idioms/option-list.md

```diff
@@ -0,0 +1,29 @@
+# Treating `Option` like a list
```

Comment:
Perhaps a better framing of this idiom is "Iterating an Option", you could then reference ref_slice as an added extra.